lib/elixir/pages/Guards.md

```diff
@@ -0,0 +1,226 @@
+# Guards
+
+Guards are a way to augment pattern matching in a predefined set of constructs.
+
+Let's see an example of a guard used in a function clause:
+
+
+```elixir
+def long_list?(list) when length(list) > 3,  do: true
```

Comment:
I don't like this example because calling length(list) is generally a bad idea altogether. Matching on `[_, _, _, _ | _]` is better than calling `length` because length will traverse the whole list, even if it has one million elements.